lms/djangoapps/commerce/api/v1/views.py

```diff
@@ -64,10 +73,17 @@ class OrderView(APIView):
     """ Retrieve order details. """
 
     authentication_classes = (SessionAuthentication,)
-    permission_classes = (IsAuthenticated,)
+    permission_classes = (IsAuthenticatedOrActivationOverridden,)
```

Comment:
One additional thing to consider is that the receipt page that is broken was specifically for use by Otto. The shopping cart has it's own separate receipt page. The only real use case for this code is if someone is using Otto, but has not enabled the flag to use the Otto receipt page (which is a deprecated use case that probably is not being utilized by anyone).